site/source/docs/porting/connecting_cpp_and_javascript/embind.rst

```diff
@@ -344,31 +349,95 @@ The JavaScript code does not need to worry about lifetime management.
 Advanced class concepts
 =======================
 
+.. _embind-object-ownership:
+
+Object Ownership
+----------------
+
+JavaScript and C++ have very different memory models which can lead to it being
+unclear which language owns and is responsible for deleting an object when it
+moves between languages. To make object ownership more explicit, *embind*
+supports smart pointers and return value policies. Return value
+polices dictate what happens to a C++ object when it is returned to JavaScript.
+
+To use a return value policy, pass the desired policy into function or method
+bindings. For example:
+
+.. code:: cpp
+
+    EMSCRIPTEN_BINDINGS(module) {
+      function("createData", &createData, return_value_policy::take_ownership());
+    }
+
+Embind supports three return value policies that behave differently depending
+on the return type of the function. The policies work as follows:
+
+* default (no argument) - Use the objects copy constructor since this is generally
+  safe and decouples returned values lifetime from the original.
```

Comment:
I do not understand what this means. We use the copy constructor, but then what happens to the object? There is no C++ code for it to live in, so it seems like RAII would immediately clean it up..?

Reply:
The copy is a new allocation, so JS gets a pointer and is then responsible for deleting the object. The value/reference that the new object was created from will be automatically deleted by C++. 

I'll reword this comment and add some more detail.